HeterogeneousCore/CUDAUtilities/interface/MessageLogger.h

```diff
@@ -0,0 +1,98 @@
+#ifndef HeterogeneousCore_CUDAUtilities_interface_MessageLogger_h
+#define HeterogeneousCore_CUDAUtilities_interface_MessageLogger_h
+
+#include <sstream>
+#include <string>
+
+namespace cudautils {
+
+  class MessageLogger {
+  public:
+    MessageLogger(std::string const& category) : category_(category) {}
+
+    MessageLogger(std::string&& category) : category_(std::move(category)) {}
+
+    ~MessageLogger() = default;
+
+    MessageLogger(MessageLogger const&) = delete;
+    MessageLogger(MessageLogger &&) = delete;
+    MessageLogger & operator=(MessageLogger const&) = delete;
+    MessageLogger & operator=(MessageLogger &&) = delete;
+
+    template <typename T>
+    MessageLogger& operator<<(T const& element) {
+      message_ << element;
+      return *this;
+    }
+
+  protected:
+    std::string category_;
+    std::stringstream message_;
```

Comment:
Note: storing the `MessageLogger` object instead of a `std::stringstream` would probably be more efficient - but then it would expose the `MessageLogger` object to the CUDA code, defeating the purpose of the wrapper.